learn/using-contracts/cw20.md

```diff
@@ -0,0 +1,224 @@
+tokens (or whatever better name you invented), but if you closed it down and
+come back, here's how to re-connect:
+
+```ts
+const client = await useOptions(coralnetOptions).setup(YOUR_PASSWORD_HERE);
+const cw20 = CW20(client);
+
+// if you forgot your address, but remember your label, you can find it again
+const contracts = await client.getContracts(4)
+contracts
+const contractAddress = contracts.filter(x => x.label === 'MINE')[0].address;
+
+// otherwise, you can just cut and paste from before
+const contractAddress = "coral13nkgqrfymug724h8pprpexqj9h629sa3yzcqfk"
+
+// now, connect to that contract and make sure it is yours
+const mine = cw20.use(contractAddress);
+mine.token_info()
+mine.minter()
+mine.balance()
+```
+
+Okay, you are connected to your contract. Let's DO SOMETHING.
+Here I will show you how you can mint tokens (you did give yourself
+that special permission on init, right?) and transfer tokens to other
+users.
+
+```ts
+const someone = "coral1hzllnaf9tezv578p3ysmml6j00ysdac4xwly9w";
+const other = "coral1dyual04q9m3f0a7qsvfx4hvu8ceyrxw9yldp8w";
+
+// right now, only you have tokens
+mine.balance()
+mine.balance(someone)
+mine.balance(other)
+// and watch the total
+mine.token_info()
+
+// let's mint some tokens for someone
+mine.mint(someone, "999888000")
+// Bonus, take the tx hash printed out and cut-paste that into https://bigdipper.coralnet.cosmwasm.com
+// eg 26D5514CF437EE584793768B56CB4E605F1F6E61FC0123030DC64E08E2EE97FA
+
+// See balances updated
+mine.balance(someone)
+mine.balance()
+// and the supply goes up
+mine.token_info()
+
+// Okay, now let's transfer some tokens... that is the more normal one, right?
+mine.transfer(other, "4567000");
+// eg. 4A76EFFEB09C82D0FEB97C3B5A9D5BADB6E9BD71F4EF248A3EF8B232C2F7262A
+mine.balance(other)
+mine.balance()
+```
+
+Great, you are moving stuff around and see it in your queries and in the block explorer.
+Time to act like a pro.
+
+## Interactive Discovery
+
+So far you have been cut-and-pasting commands I have been telling you. But what else
+can you do with this contract? Luckily Javascript has some nice introspection. And we
+added some extra bonuses there. Just type `mine` in the REPL and see the list of methods:
+
+```
+>> mine
+{ contractAddress: 'coral13nkgqrfymug724h8pprpexqj9h629sa3yzcqfk',
+  balance: [AsyncFunction: balance],
+  allowance: [AsyncFunction: allowance],
+  token_info: [AsyncFunction: token_info],
+  minter: [AsyncFunction: minter],
+  mint: [AsyncFunction: mint],
+  transfer: [AsyncFunction: transfer],
+  burn: [AsyncFunction: burn],
+  increaseAllowance: [AsyncFunction: increaseAllowance],
+  decreaseAllowance: [AsyncFunction: decreaseAllowance],
+  transferFrom: [AsyncFunction: transferFrom] }
+```
+
+But how do I call them? What arguments do they take?
+There is a great `.type` operator. Unfortunately it doesn't work
+with members directly, but with a local variable, we can get the info.
+
+This is not what we want:
```

Comment:
Telling negative results before showing code is a bit confusing. The reader will be like: **then why do you show this to me?**. Swap the lines of the code example below and this line.

Reply:
Good idea. After doing the proper way, you will think "why so long, this is a shortcut".
I tried the first one first, but yeah, I guess you want the answer first